### DISCUSSION TITLE
Run JS function on client before full content loading

### DISCUSSION THREAD

**Discussion by Jakub-Nagy** [vercel/next.js#36761] [Help] [answered]
Hi,
I have kind of an odd situation. I am using dynamic scaling on my site, the font sizes are changing smoothly based on window size. Simply using "vw" or "vh" is not enough, because zooming capability is lost. Because of this, I have implemented this snippet of code to change the rem (html) font size based on window size and zoom:
```js
// Get page zoom
let zoom = Math.round((((window.outerWidth - 10) / window.innerWidth) * 100) / 5) / 20;

// Clamp zoom
if (zoom > 1.25) zoom = 1.25;
if (zoom < 0.5) zoom = 0.5;

// Get vmin absolute size
const vminabsolute = window.innerHeight / 100;

console.log(vminabsolute * zoom);

// Set rem to vmin absolute compensating for the zoom factor
document.documentElement.style.setProperty("font-size", `${vminabsolute * zoom}px`);
```

This actually works well on resize, the problem is if I run this in React in a useEffect, all the content (along with CSS) loads first with the wrong scaling, and then 'jumps' to the correct size.

If you look on the snippet though, it isn't depended on any content or react being loaded. It just accesses the window and document objects. So this issue can be fixed with plain CSR rendered React by just putting the script right into index.html.

When using SSR, there's no statically defined index.html so all the content loads and just then it resizes. I tried using the Next <Script>  tag but then it tried running it on the server.

So the question is, can I run the above script before all the content is loaded? Perhaps even after the HTML is loaded, but at least before all the assets are loaded. Because this way Next makes the content seem buggy. Thanks.

**Comment by icyJoseph** [comment, marked as answer]
You could place this at the top of `_app.tsx`, like so:


```tsx
// _app.tsx
import type { AppProps } from "next/app";

if (typeof window !== "undefined") {
  document.addEventListener("DOMContentLoaded", () => {
    // Get page zoom
    let zoom =
      Math.round((((window.outerWidth - 10) / window.innerWidth) * 100) / 5) /
      20;

    // Clamp zoom
    if (zoom > 1.25) zoom = 1.25;
    if (zoom < 0.5) zoom = 0.5;

    // Get vmin absolute size
    const vminabsolute = window.innerHeight / 100;

    console.log(vminabsolute * zoom);

    // Set rem to vmin absolute compensating for the zoom factor
    document.documentElement.style.setProperty(
      "font-size",
      `${vminabsolute * zoom}px`
    );
  });
}

function MyApp({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />;
}

export default MyApp;
```

Also an inline script should work, but I think the above might be slightly better.

**Reply by Jakub-Nagy** [reply]
Amazing, thank you! This helped resolve the buggy behavior as well as improve the overall performance.
